src/coreclr/src/gc/gc.cpp

```diff
@@ -5566,10 +5572,32 @@ bool gc_heap::virtual_commit (void* address, size_t size, gc_oh_num oh, int h_nu
     if (heap_hard_limit)
     {
         check_commit_cs.Enter();
-        committed_by_oh[oh] += size;
         bool exceeded_p = false;
 
-        if ((current_total_committed + size) > heap_hard_limit)
+        if (heap_hard_limit_soh)
+        {
+            if (oh == gc_oh_num::soh)
+            {
+                if (committed_by_oh[oh] + size > heap_hard_limit_soh)
+                {
+                    exceeded_p = true;
+                }
+            }
+            else if (oh == gc_oh_num::loh)
```

Comment:
nit - the more I look at this code the more I wish there wasn't so much duplicated code. it seems like instead of having 3 separated values you could just have a heap_hard_limit_by_oh[3] array....and then you can avoid writing the code 3 times here and in initialization... that would seem nicer...